dashboard/app/controllers/home_controller.rb

```diff
@@ -86,14 +86,20 @@ def init_homepage
       if current_user.teacher?
         base_url = CDO.code_org_url('/teacher-dashboard#/sections/')
         @sections = current_user.sections.map do |section|
+          script = nil
           if section.script_id
-            course_name = Script.get_from_cache(section.script_id)[:name]
+            script = Script.get_from_cache(section.script_id)
+          end
```

Comment:
May or may not be worth a comment explaining why this logic exists, e.g., to guard against the section being assigned to a script not found in the cache.